src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1401,15 +1488,20 @@ NetworkOPsImp::apply(std::unique_lock<std::mutex>& batchLock)
                     << "Transaction is now included in open ledger";
                 e.transaction->setStatus(INCLUDED);
 
+                // Pop as many "reasonable" transactions for this account as
+                // possible. "Reasonable" means they have sequential sequence
+                // numbers, or use tickets.
                 auto const& txCur = e.transaction->getSTransaction();
-                auto const txNext = m_ledgerMaster.popAcctTransaction(txCur);
-                if (txNext)
+                auto txNext = m_ledgerMaster.popAcctTransaction(txCur);
+                while (txNext)
```

Comment:
To clarify, this loop doesn't do anything with the `TxQ`. This is reading the list of held transactions in `LedgerMaster`. So, AFAIK, there is no limit. However, the only way for a tx to get into that list is in this same function (see calls to `addHeldTransaction`), and only transactions received directly from RPC or from peers get to this function. That said, a malicious client or peer could probably get a bunch of transactions held (this is true without these changes). Thus, it might make sense to add a limit to mitigate some kinds of attacks.

Also, `submit_held` is a `std::vector`. Anything added to it will be processed in the _next_ batch of transactions.